esp/templates/program/modules/programprintables/studentschedule.tex

```diff
@@ -1,88 +1,99 @@
-\graphicspath{% templatetag openbrace %}{% templatetag openbrace %}{{ PROJECT_ROOT }}templates/program/modules/programprintables/{% templatetag closebrace %}{% templatetag closebrace %}
-\graphicspath{% templatetag openbrace %}{% templatetag openbrace %}{{ MEDIA_ROOT }}uploaded/{% templatetag closebrace %}{% templatetag closebrace %}
-
-\pagestyle{empty}
-\renewcommand*\arraystretch{1.5}
-\newcolumntype{+}{>{\global \let \currentrowstyle \relax }}
-\newcolumntype{^}{>{\currentrowstyle }}
-\newcommand{\rowstyle}[1]{\gdef \currentrowstyle{#1} #1\ignorespaces}
-\setlength{\parindent}{0pt}
-\setlength{\parskip}{0pt}
+\usepackage[code=Code39,X=.5mm,ratio=2.25,H=2cm]{makebarcode}
+
+\pagestyle{fancy}
+\fancyhead[C]{\Huge \textsf{\textbf{ {{ program.niceName }} } }}
+\setlength\headheight{16pt}
+\renewcommand{\headrulewidth}{3pt}
+\renewcommand{\footrulewidth}{0pt}
+\fancyfoot{}
+
+\newcommand{\nameblock}{}
+\newcommand{\barcodeblock}{}
 
 \begin{document}
 {% for student in students %}
-~\\
-\vspace*{0.25in}
+\renewcommand{\nameblock}{
+\textbf{Name: {{ student.name|texescape }} \\
+Username: {{ student.username|texescape }} \\
+User ID: {{ student.id }} \\
+}
+}
+
+\renewcommand{\barcodeblock}{
+\barcode{% templatetag openbrace %}{{ student.id }}{% templatetag closebrace %}
+}
+
+\begin{multicols}{3}
+
+\begin{flushleft}
+\nameblock
+\vspace{1ex} \hrule \vspace{1ex}
+\textbf{Amount Owed: \fbox{\${{ student.itemizedcosttotal|floatformat:"-2" }} } }
+\end{flushleft}
+
+\columnbreak
 \begin{center}
-\begin{tabularx}{17cm}{X r}
-\noindent
-{% spaceless %}
-\begin{minipage}[b]{6in}
-\Huge\textbf{% templatetag openbrace %}{{ student.first_name|texescape }} {{ student.last_name|texescape }} $\bullet$ Grade {{ student.getGrade|texescape }} } \\
- \Large \textbf{ User name: {{ student.username|texescape }} $\bullet$ ID: {{ student.id }} } \\
-\end{minipage}
-{% endspaceless %}
-&
-%\includegraphics[width=0.6in]{splogol}
-\vspace{0.25in}
+Student Schedule for \\
+
+\begin{spacing}{1}
+\LARGE \bfseries {{ student.name|texescape }}
+\end{spacing}
 
-\end{tabularx}
 \end{center}
 
+\columnbreak
+
+\begin{flushright}
+\barcodeblock
+\end{flushright}
 
+\end{multicols}
 
-\normalsize
+\renewcommand\tabularxcolumn[1]{m{#1}}
 
 \begin{center}
-\begin{tabularx}{17cm}{|X|p{1.5in}|m{3.63in}|}
+{\large \textbf{Class Schedule} \\}
+\vspace{0.25cm}
+\begin{tabularx}{\textwidth}{|m{1.7in}|X|m{0.55in}|}
 \hline
-\multicolumn{3}{c}{
-\rowstyle{\color{white}\cellcolor{black} } 
-\textbf{Class Schedule for {{ student.name|texescape }} } }  \\
-\multicolumn{1}{|c|}{\textbf{Class Time} } & \textbf{Room} & \multicolumn{1}{c|}{\textbf{Class} } \\
+\textbf{Time and Room} & \textbf{Class and Teacher} & \textbf{Code} \\
 {% for cls in student.classes %}\hline
-{% ifequal cls.event_type.description "Compulsory" %}
-  \multicolumn{1}{|c|}{ {{ cls.start|date:"f" }} -- {{ cls.end|date:"f A" }} } & \multicolumn{2}{|c|}{ \textbf{\textit{ {{ cls.description }} } } }
```

Comment:
This deleted code, are we sure no one is using this "Compulsory" logic anymore? It seems to print the full description rather than teachers/room/class title. I don't know what it's for but I'm guessing it would be used for e.g. an "opening ceremony" or similar event that all students attended before going to classes.

Reply:
Hmm...we'll still show them in the new format, so I don't think we're losing anything unless the description is really long? I guess I could add it back in and show the full description without teachers?

Reply:
The difference is that the old code is printing `cls.description` but it seems your new code is not printing `cls.description` at all (not even a truncated version of it).

Reply:
Ah, you are totally correct. What if I put the description where the teacher info goes if the class is of the "Compulsory" event type?

Reply:
That sounds like the right thing to do.

Reply:
Ok, after some digging, I now see what's going on. In this case, we inject timeslots into the class list if they are compulsory (because we don't have classes for them). That explains why we don't need to grab `cls.parent_class.get_meeting_times.0` here to check the `event_type`. Looking at other instances of where we check for whether a class is "Compulsory", it actually looks like we we use the `short_description` and/or `description` from the timeslot:
https://github.com/learning-unlimited/ESP-Website/blob/ed0733c3c797c1fc56bb4d556b9bf72b4acf848b/esp/templates/users/student_schedule.html#L104-L106
`short_description` is the timeblock information, which we already have in the first column, so, I've just replaced the class title with the `description` and the teacher line will be blank.

Reply:
Ok, looks good.